clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -156,7 +157,11 @@ object UncommittedGarbageCollector {
 
         var uncommittedDF =
           if (uncommittedGCRunInfo.uncommittedLocation != "") {
-            spark.read.parquet(uncommittedGCRunInfo.uncommittedLocation)
+            val uncommittedLocation =
```

Comment:
Can you explain what exactly was the issue with Azure?

Reply:
When we use Hadoop's FileSystem for the repository storage namespace we are getting AbstractFileSystem that can't be used for listing.
By converting the storage namespace address to `abfs://...` scheme and query for FileSystem, we are getting AzureBlobStoreFileSystem that enables the API we use.
Used the same method that the current GC users - fetch the right storage NS address based on the client you use.